tests/qnodes/test_qnode_base.py

```diff
@@ -1500,3 +1500,29 @@ def circuit(a, b, c, d, *args):
         # the user will evaluate the QNode with.
         node.set_trainable_args({0, 1, 6})
         assert node.get_trainable_args() == {0, 1, 6}
+
+
+def test_old_qnode_in_tape_mode():
+    """Test that the old QNode can still be evaluated when running in tape mode"""
+
+    # tape mode should not be active so that we can use the old QNode
+    assert not qml.tape_mode_active()
+
+    try:
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def f(x):
+            qml.RX(x, wires=0)
+            qml.RY(0.4, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        qml.enable_tape()
+        f(0.4)
```

Comment:
Might be worth to check the output here. Just to make sure it's not doing something weird (like returning `None`).

Reply:
Great, I just added a check against the hardcoded expected result: cea36c5cd89c9963f8b9dcf1c05435874a13a432